fastlane/download_metadata.swift

```diff
@@ -57,6 +57,22 @@ let languages = [
     "zh-Hant": "zh-tw",
 ]
 
+func shouldUpdateSubtitle(locale: String) -> Bool {
+    return isIn20211210Experiment(locale: locale) == false
+}
+
+func shouldUpdateKeywords(locale: String) -> Bool {
+    return isIn20211210Experiment(locale: locale) == false
+}
```

Comment:
I originally wrote a method for the app name too, but the code doesn't touch it.

I guess at the time of doing all this work, it wasn't considered we might change app name 😄 . I didn't do any more work than necessary because:

1. This is a PR against a release branch, so let's keep it nice and short to limit footprint and chances of breaking stuff
2. All this tooling will change soon thanks to @AliSoftware's ongoing improvements work

Reply:
On that note, I'm interested to hear your feedback on how we should handle this in the release toolkit.

My original plan was just to port the existing logic of that script to an action – and in fact there's already an `gp_downloadmetadata` action existing, and which [is used by WPAndroid's Fastfile](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/fastlane/Fastfile#L662-L671), so I planned to just reuse it… but in its current form it doesn't support providing exceptions for just certain metadata (i.e. just keywords and subtitle but not title nor release notes) and just certain locales (`en-*`) like you're doing here.

Given I also don't realy like [the hacky way](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/fastlane/Fastfile#L663-L664) the source locale is handled by that action, I've also considered deprecating the current `gp_downloadmetadata` action and make a new one that would be very similar but with a better API and handling of the source locale.
So that could be the occasion to allow us to provide exceptions as well like you did in the `.swift` script here, but I'm not sure of how the API/ConfigItems should look like for such a new action in order to be flexible enough while trying not to overcomplicate the API.

Reply:
I like the idea of writing a new action (or modifying the current one and rely on SemVer 🤷‍♂️ ) to allow for exceptions such as this. I don't have a formed idea of how to do it, but I'm guessing something along the line of an optional block, or array of blocks, with locale and key name as input and `Bool` as output 🤔

Reply:
Mmmh a block taking locale + key is a smart option to add flexibility here.

The tricky case when passing keys here is that the key for the release notes changes on each version (`release_notes_NNN`, with NNN changing), so it makes it not trivial (at `Fastfile`'s call site) to write an exception block that would handle those (though admittedly having an exception on the release notes key is the  least likely of the cases we'd encounter, the title, subtitle and keywords being the more likely instead)

That led me to ponder if we shouldn't just use `release_notes` (without the `_NNN` suffix) as a key (and also, stop always keeping the release notes for the previous release `NNN-1` around in GlotPress). My guess for the reason why we started doing that in the past is for rare cases where we'd do the code freeze of `NNN` before the release of `NNN-1` would be finalized (e.g. if AppStore rejected the build for `NNN-1` but we didn't want to postpone the code-freeze for `NNN` because of it), so I guess there's still a valid reason why we've been doing this and a reason to still keep that one around. But that makes me think if we should brainstorm on this as well while working on all of this.